Makefile

```diff
@@ -282,6 +296,76 @@ web-serve: web-pre-process $(HUGO)
 	@echo ">> serving documentation website"
 	@cd $(WEB_DIR) && $(HUGO) --config hugo.yaml -v server
 
+JSONNET_CONTAINER_CMD:=docker run --rm \
+		-u="$(shell id -u):$(shell id -g)" \
+		-v "$(shell go env GOCACHE):/.cache/go-build" \
+		-v "$(PWD):/go/src/github.com/thanos-io/thanos:Z" \
+		-w "/go/src/github.com/thanos-io/thanos" \
+		-e USER=deadbeef \
+		-e GO111MODULE=on \
+		quay.io/coreos/jsonnet-ci
+
+.PHONY: mixin-generate-in-container
+mixin-generate-in-container:
+	@echo ">> Compiling and generating thanos-mixin"
+	$(JSONNET_CONTAINER_CMD) make $(MFLAGS) jsonnet-vendor
+	$(JSONNET_CONTAINER_CMD) make $(MFLAGS) jsonnet-format
+	$(JSONNET_CONTAINER_CMD) make $(MFLAGS) examples
+
+.PHONY: examples
+examples: examples/alerts/alerts.md examples/alerts/alerts.yaml examples/alerts/rules.yaml examples/dashboards examples/tmp
+	$(EMBEDMD) -w examples/alerts/alerts.md
+	$(EMBEDMD) -w mixin/thanos/README.md
+
+.PHONY: examples/tmp
+examples/tmp:
+	-rm -rf examples/tmp/
+	-mkdir -p examples/tmp/
+	$(JSONNET) -J ${JSONNET_VENDOR_DIR} -m examples/tmp/ ${MIXIN_ROOT}/separated_alerts.jsonnet | xargs -I{} sh -c 'cat {} | $(GOJSONTOYAML) > {}.yaml; rm -f {}' -- {}
+
+.PHONY: examples/dashboards
+examples/dashboards: $(JSONNET) ${MIXIN_ROOT}/mixin.libsonnet ${MIXIN_ROOT}/defaults.libsonnet ${MIXIN_ROOT}/dashboards/*
+	-rm -rf examples/dashboards/*.json
+	-mkdir -p examples/dashboards/
+	$(JSONNET) -J ${JSONNET_VENDOR_DIR} -m examples/dashboards ${MIXIN_ROOT}/dashboards.jsonnet
+
+examples/alerts/alerts.yaml: $(JSONNET) $(GOJSONTOYAML) ${MIXIN_ROOT}/mixin.libsonnet ${MIXIN_ROOT}/defaults.libsonnet ${MIXIN_ROOT}/alerts/*
+	$(JSONNET) ${MIXIN_ROOT}/alerts.jsonnet | $(GOJSONTOYAML) > $@
+
+examples/alerts/rules.yaml: $(JSONNET) $(GOJSONTOYAML) ${MIXIN_ROOT}/mixin.libsonnet ${MIXIN_ROOT}/defaults.libsonnet ${MIXIN_ROOT}/rules/*
+	$(JSONNET) ${MIXIN_ROOT}/rules.jsonnet | $(GOJSONTOYAML) > $@
+
+.PHONY: jsonnet-vendor
+jsonnet-vendor: $(JSONNET_BUNDLER) jsonnetfile.json jsonnetfile.lock.json
+	rm -rf ${JSONNET_VENDOR_DIR}
+	$(JSONNET_BUNDLER) install --jsonnetpkg-home="${JSONNET_VENDOR_DIR}"
+
+JSONNET_FMT := jsonnetfmt -n 2 --max-blank-lines 2 --string-style s --comment-style s
+
+.PHONY: jsonnet-format
+jsonnet-format:
+	find . -name 'vendor' -prune -o -name '*.libsonnet' -print -o -name '*.jsonnet' -print | \
+		xargs -n 1 -- $(JSONNET_FMT) -i
+
+.PHONY: jsonnet-format-in-container
+jsonnet-format-in-container:
+	$(JSONNET_CONTAINER_CMD) make $(MFLAGS) jsonnet-format
+
+.PHONY: rules-lint
+rules-lint: $(PROMTOOL) examples/alerts/alerts.yaml examples/alerts/rules.yaml
+	$(PROMTOOL) check rules examples/alerts/alerts.yaml examples/alerts/rules.yaml
+
+.PHONY: alerts-test
+alerts-test: rules-lint
+	$(PROMTOOL) test rules examples/alerts/tests.yaml
+
+.PHONY: examples-clean
+examples-clean:
```

Comment:
It might be beneficial to actually run it always instead of relying on Makefile `run if changed`. Then we can put all in a single `example, example-in-container` and `example-rules-lint` targets. The process is super fast anyway .. Not a strong opinion, up to you (:

Reply:
But since you already did hard work let's keep it this way nice one!